server/services/store/sqlstore/data_migrations.go

```diff
@@ -0,0 +1,58 @@
+package sqlstore
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/mattermost/focalboard/server/model"
+	"github.com/mattermost/focalboard/server/utils"
+)
+
+const (
+	UniqueIDsMigrationKey = "UniqueIDsMigrationComplete"
+)
+
+func (s *SQLStore) runUniqueIDsMigration() error {
+	setting, err := s.GetSystemSetting(UniqueIDsMigrationKey)
+	if err != nil {
+		return fmt.Errorf("cannot get migration state: %w", err)
+	}
+
+	// If the migration is already completed, do not run it again.
+	if hasAlreadyRun, _ := strconv.ParseBool(setting); hasAlreadyRun {
+		return nil
+	}
+
+	s.logger.Debug("Running Unique IDs migration")
+
+	blocks, err := s.GetBlocksWithSameID()
```

Comment:
I would expect here to use the private method and also to do everything in the same transaction, to make it 1 single all or nothing. Also it is going to be more performant. And because this is being executed as part of the migrations process is going to be the only thing running at that time, so I prefer it more performant and atomic.

Reply:
Addressed in 7f1da7c